tck/features/ListOperations.feature

```diff
@@ -0,0 +1,748 @@
+      """
+    Then the result should be:
+      | res   |
+      | true  |
+    And no side effects
+
+  Scenario: IN should return null if comparison with null is required for empty list
+    When executing query:
+      """
+      RETURN [] IN [1, 2, null] AS res
+      """
+    Then the result should be:
+      | res   |
+      | null  |
+    And no side effects
+
+  Scenario: IN should return true when LHS and RHS contain nested list with multiple empty lists
+    When executing query:
+      """
+      RETURN [[], []] IN [1, [[], []]] AS res
+      """
+    Then the result should be:
+      | res   |
+      | true  |
+    And no side effects
+  # Equality
+
+  Scenario: Equality between list and literal should return false
+    When executing query:
+      """
+      RETURN [1, 2] = 'foo' AS res
+      """
+    Then the result should be:
+      | res   |
+      | false  |
+    And no side effects
+
+  Scenario: Equality of lists of different length should return false despite nulls
+    When executing query:
+      """
+      RETURN [1] = [1, null] AS res
+      """
+    Then the result should be:
+      | res   |
+      | false  |
+    And no side effects
+
+  Scenario: Equality between different lists with null should return false
+    When executing query:
+      """
+      RETURN [1, 2] = [null, 'foo'] AS res
+      """
+    Then the result should be:
+      | res   |
+      | false  |
+    And no side effects
+
+  Scenario: Equality between almost equal lists with null should return null
+    When executing query:
+      """
+      RETURN [1, 2] = [null, 2] AS res
+      """
+    Then the result should be:
+      | res   |
+      | null  |
+    And no side effects
+
+  Scenario: Equality of nested lists of different length should return false despite nulls
+    When executing query:
+      """
+      RETURN [[1]] = [[1], [null]] AS res
+      """
+    Then the result should be:
+      | res    |
+      | false  |
+    And no side effects
+
+  Scenario: Equality between different nested lists with null should return false
+    When executing query:
+      """
+      RETURN [[1, 2], [1, 3]] = [[1, 2], [null, 'foo']] AS res
+      """
+    Then the result should be:
+      | res    |
+      | false  |
+    And no side effects
+
+  Scenario: Equality between almost equal nested lists with null should return null
+    When executing query:
+      """
+      RETURN [[1, 2], ['foo', 'bar']] = [[1, 2], [null, 'bar']] AS res
+      """
+    Then the result should be:
+      | res    |
+      | null  |
+    And no side effects
+
+  # General
+
+  Scenario: Return list size
+    When executing query:
+      """
+      RETURN size([1, 2, 3]) AS n
+      """
+    Then the result should be:
+      | n |
+      | 3 |
+    And no side effects
+
+  Scenario: Setting and returning the size of a list property
+    And having executed:
+      """
+      CREATE (:Label)
+      """
+    When executing query:
+      """
+      MATCH (n:Label)
```

Comment:
This scenario needs a different `Given` step. It may give more rows if given a graph with `:Label`-labeled nodes in it, which is allowed by `any graph`.

Each scenario may only have one `Given`, and the `Background` is overridden by what the scenario itself says, so we could just say `Given an empty graph` here.